src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -180,6 +180,7 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
         return;
     }
 
+    slotPending();
```

Comment:
I think the real issue is the the base class unconditionally sets the state to finished even if it is not sure.
I think that should be fixed there:
https://github.com/mixxxdj/mixxx/blob/f2d670edeb483efb066bf7f00255d01df2f0017c/src/network/webtask.cpp#L395

How about returning true = finished and false = expect more and only set the state in the first case.
Without it was pending before anyway, am I right?

Reply:
I didn't fully understand. Should I change the WebTask to check if there are any other upcoming tasks? If there are any more tasks, finished is changed to false and if there is none left finished becomes true?

Reply:
Before the last commit normally what was happening:

The task was finished, so aborting the task (change the track, closing the tag fetcher) was returning debug assert, and the error message was `Cannot abort terminated task`. 

So, I wanted to add external state, I changed to Pending. But this time, after I call the slotStart it was returning `Task is still busy and cannot be started again`, so I added another external state, Idle. And it worked just fine but I am not sure if that was the correct approach.

Reply:
The approach sound reasonable. It is just that the state is private and your functions expose it. 

my id has to make a  
```
bool MusicBrainzRecordingsTask::doNetworkReplyFinished()
```

and than in WebTask 

```
if (doNetworkReplyFinished()) {
   state = finished;
}
```

Reply:
this might be extended to

```
else {
  state = >waiting for another request< 
}
```
The assertion can be than changed to assert either finish or waiting for another request ...

Reply:
Can I change the do network reply finished to from void to bool? I am asking because it is used in MusicBrainzTask and  override in Jsonwebtask for Acoustidtask. 

Would it work fine?

Reply:
They need to be adjusted as well. I think for always return true?

Reply:
Yeap that works actually, I made a little mistake that was trying to parse the JSON two times, and that's why it was causing an error.

I think with another state as you suggested, we can find a solution to this problem.